src/tree.ml

```diff
@@ -686,18 +706,25 @@ let rec compile : CM.t -> tree = fun ({clauses ; positions ; slot} as pats) ->
       in
       (* Default child *)
       let default =
+        let compile = compile vars_id in
         let (positions, clauses) = CM.default swap positions updated in
         let ncm = CM.{clauses ; slot ; positions} in
         if CM.is_empty ncm then None else Some(compile ncm)
       in
       (* Abstraction specialisation*)
       let abstraction =
-        if List.for_all (fun (x, _) -> x <> TC.Abst) cons then None else
-        let var = Bindlib.new_var mkfree "tr" in
+        let is_abst = function Abst(_) -> true | _ -> false in
+        if List.for_all (fun x -> not (is_abst x)) column then None else
+        let var = Bindlib.new_var mkfree var_prefix in
+        let id = !count in
```

Comment:
you can avoid this let by incrementing first

Reply:
I don't think I can since, in the last line, I need the `id` variable, as well as the counter updated, for the new call to `compile`

Reply:
To be more precise, the value of the counter must be saved to be written in the resulting tuple `Some(id, compile...)`. But the call to `compile` which seems to be performed before the evaluation of the `id` variable (caml reads arguments from right to left it seems) modifies the counter.

Reply:
I see.